### PR TITLE
[Merged by Bors] - chore: move `mul_ite`, add `dite` version, multiplicativise `ite_add_ite`

### PR DESCRIPTION
**Pull request by YaelDillies**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Defs.lean [hunk 1]
```diff
@@ -168,6 +168,52 @@ class Inv (α : Type u) where
 @[inherit_doc]
 postfix:max "⁻¹" => Inv.inv
 
+section ite
+variable {α : Type*} (P : Prop) [Decidable P]
+
+@[to_additive]
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/Defs.lean:174]: A section `Mul` for these seems appropriate

File: Mathlib/Algebra/Group/Defs.lean [hunk 2]
```diff
@@ -168,6 +168,52 @@ class Inv (α : Type u) where
...
+lemma mul_dite [Mul α] (a : α) (b : P → α) (c : ¬ P → α) :
+    (a * if h : P then b h else c h) = if h : P then a * b h else a * c h := by split <;> rfl
+
+@[to_additive]
+lemma mul_ite [Mul α] (a b c : α) : (a * if P then b else c) = if P then a * b else a * c := by
+  split <;> rfl
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/Defs.lean:180]:
```suggestion
lemma mul_ite [Mul α] (a b c : α) : (a * if P then b else c) = if P then a * b else a * c := 
  mul_dite _ _ _ _
```
if you like.
Similar below

**Review by b-mehta — COMMENTED** [review]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**